[Onboarding] Wire up read data and logic

sztomek opened the pull request:
## Description
This PR adds the next piece to the puzzle: the new onboarding screen now works with actual data!

Fixes https://linear.app/a8c/issue/PCDROID-28/wire-up-logic-on-upgrade-screens

## Testing Instructions
1. Apply [debug-as-prod (1).patch](https://github.com/user-attachments/files/21287520/debug-as-prod.1.patch)
2. Build `debugProd` variant and install it
3. Do not log in upon opening the app
4. Navigate to Settings -> Pocket Casts Plus
5. Select a plan and tap 'Subscribe' or 'Start free trial' if you're eligible (depends on your google account, if you've ever claimed the free trial with that google account, you won't be eligible)
6. Go through subscription flow presented by the Play billing sdk
- [ ] things worked

## Screencast
https://github.com/user-attachments/assets/9f682269-0065-4426-9452-90facaffa64b


## Checklist
- [ ] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] ~I have considered whether it makes sense to add tests for my changes~
- [ ] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [ ] ~Any jetpack compose components I added or changed are covered by compose previews~
- [ ] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~

MiSikora reviewed (changes requested):
I couldn't test the PR. Applying the patch doesn't work.

```
git apply ~/Downloads/debug-as-prod.1.patch
error: patch failed: build.gradle.kts:319
error: build.gradle.kts: patch does not apply
```

sztomek commented:
sorry, attached the wrong patch... 
[debug-as-prod copy.patch](https://github.com/user-attachments/files/21302636/debug-as-prod.copy.patch)
please try with this one.
I have also addressed all the comments on this PR
